tests/core/pyspec/eth2spec/test/helpers/shard_block.py

```diff
@@ -0,0 +1,47 @@
+from eth2spec.test.helpers.keys import privkeys
+from eth2spec.utils import bls
+from eth2spec.utils.bls import only_with_bls
+
+
+@only_with_bls()
+def sign_shard_block(spec, beacon_state, shard, block, proposer_index=None):
+    slot = block.message.slot
+    if proposer_index is None:
+        proposer_index = spec.get_shard_proposer_index(beacon_state, slot, shard)
+
+    privkey = privkeys[proposer_index]
+    domain = spec.get_domain(beacon_state, spec.DOMAIN_SHARD_PROPOSAL, spec.compute_epoch_at_slot(slot))
+    signing_root = spec.compute_signing_root(block.message, domain)
+    block.signature = bls.Sign(privkey, signing_root)
+
+
+def build_shard_block(spec,
+                      beacon_state,
+                      shard,
+                      slot,
+                      body=None,
+                      signed=False):
+    shard_state = beacon_state.shard_states[shard]
+    if slot is None:
+        slot = shard_state.slot
```

Comment:
Should this be defaulting to `shard_state.slot + 1`? If the `beacon_state.shard_states[shard]` already has `slot` referenced, then the next block to be built should be the _next_ slot, not the same slot.

Reply:
Or maybe to `beacon_state.slot` as the normal case will be to built a shard for the same slot as the current beacon state